docs/upgrading/migrating-local-extensions-33.md

```diff
@@ -0,0 +1,138 @@
+---
+title: "Migrating Local Extensions for Bolt 3.3"
+---
+Migrating Local Extensions for Bolt 3.3
+===========================================
+
+Bolt 3.3 brings with it a fairly big refactor of local extensions and
+unavoidably this will mean some breaking changes.
+
+The good news is that by changing the approach a little we will hopefully make
+it a lot easier to write and add local extensions to your applications.
+
+## The Essential Migration Steps
+
+### Autoloading
+
+Local extensions are not autoloaded within the extensions directory anymore so
+you will need to add each local extensions directory to the `composer.json`
+file in `extensions/`.
+
+As an example, if you currently have a local extension at:
+`extensions/local/myname/myextension` your extension class may either be in that
+directory, or perhaps inside a `src` directory.
+
+Let's assume the class is in the root directory, you need to open your root composer.json
+file and add this to the autoload/psr-4 section.
+
+    "Bolt\\Extension\\Myname\\MyExtensionNamespace\\": "local/myname/myextension"
+    
+If you don't currently have a psr-4 section in your `extensions/composer.json` file, then
+add it so it looks something like this
+
+```json
+    "autoload" : {
+        "psr-4" : {
+            "Bolt\\Extension\\Myname\\MyExtensionNamespace\\": "local/myname/myextension"
+        }
+    }
+```
+
+You need to make one entry for each of your local extensions. The setup here is
+identical to how you would setup your own application and there's absolutely no
+need to keep your extensions inside `extensions/local` any more. You may well
+decide to move extensions to be part of your main App namespace and if so you
+can do something like the following: 
+
+```json
+    "autoload" : {
+        "psr-4" : {
+            "MyCompany\\MyApp\\": "src"
+        }
+    }
+```
+
+With the above setup you can move local extensions to `src/Extension/` and
+access the class via `new MyCompany\MyApp\Extension\SuperCoolExtension()`
+
+### Activation
+
+Once your extensions can be autoloaded then one more step is needed to direct
+Bolt to load them.
+
+If you have a `.bolt.yml` file at the root of your project you can define which
+extension classes are loaded within that file. If you open the file as it is now
+then it will look something like this:
+
+```yaml
+
+paths:
+    web: public
+    themebase: public/theme
+    files: public/files
+    view: public/bolt-public/view
+```
```

Comment:
I've been (sometimes gumpily) all over the place on this one since @CarsonF first started bugging me about it on one of my PRs.

In an attempt to prove my point, I looked over the docs and notices that we've had, or are developing, a habit of re-documenting tiny bits in many places leading to inconsistency in the docs.

To that end, my vote w.r.t. approach on these is to **only** show the change, but put a hyperlinked note near said blocks to the "canonical" documentation for the surrounding sections like `paths:` … strikes me as making all of our lives easier in the long-run with keeping our documentation consistent.